### PR TITLE
apps/testcase/systemio: add new spi scenarios TC

### PR DESCRIPTION
**Pull request by samegautam**
Filepath :  apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c
Description : add new scenarios testcase for SPI

Scenarios testcase : 
itc_iotbus_spi_open_close_multi_handler_p
itc_iotbus_spi_open_close_modes_p
itc_iotbus_spi_open_close_invalid_bus_n
itc_iotbus_spi_open_close_invalid_bpw_n
itc_iotbus_spi_open_close_invalid_frequecy_n

### CODE REVIEW

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by samegautam** [comment]
Please review @jeongchanKim @sunghan-chang @junmin-kim @Taejun-Kwon

File: apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c [hunk 1]
```diff
@@ -18,23 +18,17 @@
...
-#include <stdlib.h>
-#include <errno.h>
-#include <tinyara/time.h>
-#include <sys/time.h>
 #include "itc_internal.h"
 #include <iotbus_spi.h>
 #include <iotbus_error.h>
 
-iotbus_spi_context_h spi;
+unsigned int g_bus = 0;
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c]: please add "static".

File: apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c [hunk 2]
```diff
@@ -18,23 +18,17 @@
...
 
-unsigned int gBus = 0;
-
-struct iotbus_spi_config_s _g_st_config = {
+struct iotbus_spi_config_s g_st_config = {
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c]: same as previous

File: apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c [hunk 3]
```diff
@@ -61,6 +55,159 @@ void itc_iotbus_spi_open_close_p(void)
...
 
+/**
+* @testcase         itc_iotbus_spi_open_close_multi_handler_p
+* @brief            initializes and closes the spi_context with multi handler
+* @scenario         initializes and closes the spi_context with multi handler
+* @apicovered       iotbus_spi_open, iotbus_spi_close
+* @precondition     none
+* @postcondition    none
+*/
+void itc_iotbus_spi_open_close_multi_handler_p(void)
```
Comment (sunghan-chang) [inline apps/examples/testcase/ta_tc/systemio/itc/itc_spi.c]: same as previous for every local functions

**Comment by sunghan-chang** [comment]
@samegautam For my coments, you should follow it only for your changes. To adding "static" for originals should be done by another commit.

**Comment by seinfra** [comment]
Code Rule Check OK.

**Comment by samegautam** [comment]
@sunghan-chang : Please review.
In fix , we have changed testcase name as per updated coding conventions.

**Comment by samegautam** [comment]
Changes would be commit in split commit.

**Pull request closed without merging**